includes/functions.php

```diff
@@ -152,10 +152,20 @@ function tsml_calculate_attendance_option($types, $approximate)
 
 //called by register_activation_hook in 12-step-meeting-list.php
 //hands off to tsml_custom_post_types
-function tsml_change_activation_state()
+function tsml_plugin_activation()
 {
 	tsml_custom_post_types();
 	flush_rewrite_rules();
+	add_option( 'tsml_plugin_new_activation', true );
+	update_option( 'tsml_plugin_new_activation', true );
```

Comment:
couple ideas here, but the main one is: if the point of setting the option is to wait to flush the rewrite rules in the init hook, how about instead running `tsml_custom_post_types` above the flush_rewrite_rules here? then we don't need the option or the change to init…

Reply:
Gotcha was the `tsml_custom_post_types()` was already running before the `flush_rewrite_rules()`, so changing order here doesn't seem to have an impact. There's still something up with the timing, when the `activate_${plugin}` plugin action happens that the rewrite rules get picked up, but they appear broken or invalid. I started with this approach, "well, just flush rewrite on the next full page load", but yeah, it's not ideal having an option for this one step.

Will keep digging if there's a way to have it on activation all rewrite rules are good by the end of that plugin activation page request.